test/cpp/tensorexpr/test_cpp_codegen.cpp

```diff
@@ -0,0 +1,47 @@
+#include <gtest/gtest.h>
+
+#include <test/cpp/tensorexpr/test_base.h>
+
+#include "torch/csrc/jit/tensorexpr/cpp_codegen.h"
+#include "torch/csrc/jit/tensorexpr/mem_arena.h"
+#include "torch/csrc/jit/tensorexpr/stmt.h"
+
+namespace torch {
+namespace jit {
+
+using namespace torch::jit::tensorexpr;
+
+TEST(CppPrinter, AllocateOnStackThenFree) {
+  constexpr int dim0 = 2, dim1 = 3;
+  KernelScope kernel_scope;
+  VarHandle var("x", kHandle);
+  Allocate* alloc = Allocate::make(var, kInt, {dim0, dim1});
+  Free* free = Free::make(var);
+
+  std::stringstream ss;
+  CppPrinter printer(&ss);
+  printer.visit(alloc);
+  printer.visit(free);
+  ASSERT_EQ(ss.str(), "int x[6];\n");
+}
+
+TEST(CppPrinter, AllocateOnHeapThenFree) {
+  constexpr int dim0 = 20, dim1 = 50, dim2 = 3;
+  KernelScope kernel_scope;
+  VarHandle var("y", kHandle);
+  Allocate* alloc = Allocate::make(var, kLong, {dim0, dim1, dim2});
+  Free* free = Free::make(var);
+
+  std::stringstream ss;
+  CppPrinter printer(&ss);
+  printer.visit(alloc);
+  printer.visit(free);
+  // size(long) = 8;
+  // dim0 * dim1 * dim2 * size(long) = 24000.
+  ASSERT_EQ(
```

Comment:
Nit: You could use `FileCheck` for the output verification.
Here is an example:
https://github.com/pytorch/pytorch/blob/e2041ce3540a08dcdc70eaef86b13ca84b31ebd5/test/cpp/tensorexpr/test_loopnest.cpp#L3116-L3123